tests/src/core/testqgsrasterfill.cpp

```diff
@@ -96,11 +100,18 @@ void TestQgsRasterFill::initTestCase()
   mpPolysLayer->setSimplifyMethod( simplifyMethod );
 
   // Register the layer with the registry
-  QgsProject::instance()->addMapLayers(
-    QList<QgsMapLayer *>() << mpPolysLayer );
+  QgsProject::instance()->addMapLayers( QList<QgsMapLayer *>() << mpPolysLayer );
 
-  //setup raster fill
+  // Setup raster fill
   mRasterFill = new QgsRasterFillSymbolLayer();
+  mRasterFill->setImageFilePath( mTestDataDir + QStringLiteral( "sample_image.png" ) );
+  mRasterFill->setWidthUnit( QgsUnitTypes::RenderPixels );
+  mRasterFill->setWidth( 30.0 );
+  mRasterFill->setCoordinateMode( QgsRasterFillSymbolLayer::Feature );
+  mRasterFill->setOpacity( 1.0 );
+  mRasterFill->setOffset( QPointF( 0, 0 ) );
+  mRasterFill->setOffsetUnit( QgsUnitTypes::RenderPixels );
```

Comment:
Can you leave out these changes? The intention with putting them in the ::init() method is that that method gets called before every test, so if an individual test messes with the mRasterFill definition then any changes it makes are reset when the next test method is called.

Reply:
Do you mean that each new test should start with exactly the same source image defined in the ::init() ?

Reply:
It should start with new instances of the symbols. Just revert your changes, add the new test alone, and it should all work fine 😁

Reply:
Done.